src/components/AstronautsPage/actions/getAstronautsData.ts

```diff
@@ -0,0 +1,20 @@
+import { AppDispatch } from '../../../store';
+import type { Astronauts } from '../types/Astronauts';
+import { fetchAstronauts } from '../api/services/fetchAstronauts';
+import {
+  astronautsRequestAction,
+  astronautsSuccessAction,
+  astronautseErorAction,
+} from './actionCreators';
+
+export const getAstronautsData =
+  () =>
+  async (dispatch: AppDispatch): Promise<void> => {
+    dispatch(astronautsRequestAction());
+    try {
+      const response: Astronauts[] = await fetchAstronauts();
+      dispatch(astronautsSuccessAction(response));
+    } catch (error) {
+      dispatch(astronautseErorAction((error as Error).message));
```

Comment:
move type to arguments

Reply:
it is the best way to include typescript for catch block